src/rust/ide/src/double_representation/module.rs

```diff
@@ -0,0 +1,143 @@
+//! Code for module-level double representation processing.
+
+use crate::prelude::*;
+
+use crate::double_representation::definition;
+use crate::double_representation::definition::DefinitionProvider;
+
+use ast::crumbs::ChildAst;
+use ast::known;
+use enso_protocol::language_server;
+
+
+
+// ==============
+// === Errors ===
+// ==============
+
+#[allow(missing_docs)]
+#[derive(Fail,Clone,Debug)]
+#[fail(display="Cannot find method by pointer {:?}.",_0)]
+pub struct CannotFindMethod(language_server::MethodPointer);
+
+#[allow(missing_docs)]
+#[derive(Copy,Fail,Clone,Debug)]
+#[fail(display="Encountered an empty definition ID. They must contain at least one crumb.")]
+pub struct EmptyDefinitionId;
+
+
+
+// ========================
+// === Module Utilities ===
+// ========================
+
+/// Looks up graph in the module.
+pub fn traverse_for_definition
```

Comment:
IMO "look up" and "traverse" are completely different things. Traverse should allow you to run lambda on each level. Looking at the signature, this should be named `get_definition` instead (like in HashMap). The same applies to function made `lookup_...` below - which implements the same kind of logic.